### ISSUE TITLE
Exit navigation loses direction under certain conditions

### ISSUE THREAD

**Issue by pj00** [CCOMIT/CSO#6]
1. Go to [/Main/GraduateMedicalEducation/PersonGraduateMedicalEducation/](url) for someone
2. Add or Edit a training entry
3. Click the Verifying Org Select button
4. Click Add
5. Enter some stuff into the text inputs (don't click the Select buttons) and click Create
6. If you Exit at this point, everything is fine whether you Exit or Select from the next page.
7. However, if you Select a Hospital or Specialty at this point and then Exit, you'll get an error if you Exit from the next page.  If you Select a Hospital or Specialty, then Save, then Exit, if you Exit or Select from the next page it takes you back to the Verifying Org lookup instead of [/Main/GraduateMedicalEducation/Edit](url)

Basically, the problem is when you do a lookup for Hospital or Specialty the ExitController, etc, get changed so it knows to go back to the Verifying Org lookup.  But this means the original exit info is lost.

```
            if (ViewBag.SelectedId != null)
            {
                nav.AddId = ViewBag.SelectedId.ToString();
                nav.ExitArea = AppSession.ExitArea;
                nav.ExitAction = AppSession.ExitAction;
                nav.ExitController = AppSession.ExitController;
                nav.ExitId = AppSession.ExitId.ToString();
            }
```

The Education Verifying Org almost certainly suffers from the same issue.

**Comment by pj00** [comment]
Maybe it would be easier to dump Jim's method for tracking this and store a List<FormNavModel> in Session[]?  Then you could Push and Pop when each page is called, or clear based on some flag that tells you whether you just came from an expected page.
